### PR TITLE
Add error handling for empty columns to SqlAlchemyDataset

### PR DESCRIPTION
**Pull request by jpcampb2**
Also, add associated tests and docs.

### CODE REVIEW

File: docs/source/expectations.rst [hunk 1]
```diff
@@ -135,3 +135,12 @@ This is how you always know what to expect from your data.
...
 For more detail on how to control expectation output, please see :ref:`standard_arguments` and :ref:`result_format`.
 
+Types of Expectations
+------------------------------------------------------------------------------
+
+Under the hood, great_expectations evaluates similar kinds of expectations using standard logic, including:
+* `column_map_expectations`, which apply their condition to each value in a column independently of other values
+* `column_aggregate_expectations`, which apply their condition to an aggregate value or values from the column
+
+In general, if a column is empty, a column_map_expectation will return True (vacuously), whereas a column_aggregate_expectation will return false (since no aggregate value could be computed).
```
Comment (abegong) [inline docs/source/expectations.rst]: Nit: `False` > `false`

File: great_expectations/dataset/sqlalchemy_dataset.py [hunk 2]
```diff
@@ -137,7 +145,7 @@ def inner_wrapper(self, column, result_format = None, *args, **kwargs):
             if result_format['result_format'] == 'BOOLEAN_ONLY':
                 return return_obj
 
-            # Use the element and null count information from a column_map_expectation if it needed
+            # Use the element and null count information from a column_aggregate_expectation if it needed
```
Comment (abegong) [inline great_expectations/dataset/sqlalchemy_dataset.py]: Nit: `if it is needed`

File: great_expectations/dataset/sqlalchemy_dataset.py [hunk 3]
```diff
@@ -617,21 +658,30 @@ def expect_column_mean_to_be_between(self,
...
+                success = (min_value <= col_avg) and (col_avg <= max_value)
 
-        elif min_value == None and max_value != None:
-            success = (col_avg <= max_value)
+            elif min_value == None and max_value != None:
+                success = (col_avg <= max_value)
 
-        elif min_value != None and max_value == None:
-            success = (min_value <= col_avg)
+            elif min_value != None and max_value == None:
```
Comment (abegong) [inline great_expectations/dataset/sqlalchemy_dataset.py:676]: Out of curiosity: you changed the other expectations to "is not None". Why not these?

**Review by abegong — APPROVED** [review]
LGTM! I put a couple tiny nits in comments.

File: docs/source/expectations.rst (hunk 1 above)
Comment (ccnobbli) [inline docs/source/expectations.rst]: > thereefore

File: tests/column_aggregate_expectations/expect_column_median_to_be_between.json [hunk 4]
```diff
@@ -1,128 +1,170 @@
...
+            "traceback_substring": "min_value and max_value cannot both be None"
+          }
+        }
+      ]},
+    {
+      "data": {
+        "empty_column": []
+      },
+      "tests": [{
+        "title": "Test empty column",
```
Comment (ccnobbli) [inline tests/column_aggregate_expectations/expect_column_median_to_be_between.json]: "Test empty column" is a bit generic of a test description, maybe you could provide some info about the expected behavior?

File: docs/source/expectations.rst (hunk 1 above)
Comment (jpcampb2) [reply docs/source/expectations.rst]: Changed!

File: great_expectations/dataset/sqlalchemy_dataset.py (hunk 2 above)
Comment (jpcampb2) [reply great_expectations/dataset/sqlalchemy_dataset.py]: Changed!

File: docs/source/expectations.rst (hunk 1 above)
Comment (jpcampb2) [reply docs/source/expectations.rst]: Changed!

File: tests/column_aggregate_expectations/expect_column_median_to_be_between.json (hunk 4 above)
Comment (jpcampb2) [reply tests/column_aggregate_expectations/expect_column_median_to_be_between.json]: Improved description!

File: great_expectations/dataset/sqlalchemy_dataset.py (hunk 3 above)
Comment (jpcampb2) [reply great_expectations/dataset/sqlalchemy_dataset.py:676]: No good reason. My IDE (PyCharm) cites the PEP8 preference to use is/is not None, and the ones I changed probably caught my eye, but the == tests are so pervasive in the codebase, I wasn't sure I wanted to change everything without talking about it...

**Pull request merged**